### PR TITLE
lightning: avoid directly return error if send check multi ingest request failed

### LINKED ISSUE #1323: v5.1.0 - lighting - gRPC i/o timeout
lightning log:
```
[2021/07/07 13:52:06.177 +08:00] [INFO] [lightning.go:164] ["starting HTTP server"] [address=0.0.0.0:82ssxx9]
[2021/07/07 13:52:06.178 +08:00] [WARN] [config.go:518] ["currently only per-task configuration can be applied, global configuration changes can only be made on startup"] ["global config changes"="[lightning.file,lightning.level,lightning.status-addr]"]
[2021/07/07 13:52:06.182 +08:00] [INFO] [info.go:40] ["Welcome to TiDB-Lightning"] [release-version=v5.1.0] [git-hash=f2fcbfa5a9783e33ff44b696920d8bbe6a19eaa6] [git-branch=heads/refs/tags/v5.1.0] [go-version=go1.16.4] [utc-build-time="2021-06-24 07:09:26"] [race-enabled=false]
[2021/07/07 13:52:06.182 +08:00] [INFO] [lightning.go:219] [cfg] [cfg="{\"id\":1625637126182164505,\"lightning\":{\"table-concurrency\":6,\"index-concurrency\":2,\"region-concurrency\":64,\"io-concurrency\":5,\"check-requirements\":true,\"meta-schema-name\":\"lightning_metadata\"},\"tidb\":{\"host\":\"172.24.xxx8.21\",\"port\":4000,\"user\":\"superdba\",\"status-port\":10080,\"pd-addr\":\"172.24.xxx.22:12379\",\"sql-mode\":\"ONLY_FULL_GROUP_BY,NO_AUTO_CREATE_USER\",\"tls\":\"false\",\"security\":{\"ca-path\":\"\",\"cert-path\":\"\",\"key-path\":\"\",\"redact-info-log\":false},\"max-allowed-packet\":67108864,\"distsql-scan-concurrency\":15,\"build-stats-concurrency\":20,\"index-serial-scan-concurrency\":20,\"checksum-table-concurrency\":2},\"checkpoint\":{\"schema\":\"tidb_lightning_checkpoint\",\"driver\":\"file\",\"enable\":true,\"keep-after-success\":false},\"mydumper\":{\"read-block-size\":65536,\"batch-size\":107374182400,\"batch-import-ratio\":0,\"data-source-dir\":\"file:///hbdata_backup/dumpfiles_4013\",\"character-set\":\"auto\",\"csv\":{\"separator\":\",\",\"delimiter\":\"\\\"\",\"null\":\"\\\\N\",\"header\":true,\"trim-last-separator\":false,\"not-null\":false,\"backslash-escape\":true},\"max-region-size\":268435456,\"filter\":[\"*.*\",\"!mysql.*\",\"!sys.*\",\"!INFORMATION_SCHEMA.*\",\"!PERFORMANCE_SCHEMA.*\",\"!METRICS_SCHEMA.*\",\"!INSPECTION_SCHEMA.*\"],\"files\":null,\"no-schema\":true,\"case-sensitive\":false,\"strict-format\":false,\"default-file-rules\":true,\"ignore-data-columns\":null},\"tikv-importer\":{\"addr\":\"\",\"backend\":\"local\",\"on-duplicate\":\"replace\",\"max-kv-pairs\":4096,\"send-kv-pairs\":32768,\"region-split-size\":100663296,\"sorted-kv-dir\":\"/hbdata_sort/sort\",\"**disk-quota\":7150160138240**,\"range-concurrency\":16,\"engine-mem-cache-size\":536870912,\"local-writer-mem-cache-size\":134217728},\"post-restore\":{\"checksum\":\"required\",\"analyze\":\"optional\",\"level-1-compact\":false,\"post-process-at-last\":true,\"compact\":false},\"cron\":{\"switch-mode\":\"5m0s\",\"log-progress\":\"5m0s\",\"check-disk-quota\":\"1m0s\"},\"routes\":null,\"security\":{\"ca-path\":\"\",\"cert-path\":\"\",\"key-path\":\"\",\"redact-info-log\":false},\"black-white-list\":{\"do-tables\":null,\"do-dbs\":null,\"ignore-tables\":null,\"ignore-dbs\":null}}"]
[2021/07/07 13:52:06.203 +08:00] [INFO] [lightning.go:282] ["load data source start"]
[2021/07/07 13:52:06.718 +08:00] [INFO] [loader.go:287] ["[loader] file is filtered by file router"] [path=metadata]
[2021/07/07 13:52:17.563 +08:00] [INFO] [lightning.go:285] ["load data source completed"] [takeTime=11.35913595s] []
[2021/07/07 13:52:17.563 +08:00] [INFO] [checkpoints.go:979] ["open checkpoint file failed, going to create a new one"] [path=/tmp/tidb_lightning_checkpoint.pb] [error="open /tmp/tidb_lightning_checkpoint.pb: no such file or directory"]
[2021/07/07 13:52:18.583 +08:00] [ERROR] [lightning.go:307] ["restore failed"] [error="build local backend failed: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 172.24.xxx44:20xx0: i/o timeout\""]
[2021/07/07 13:52:18.583 +08:00] [ERROR] [main.go:90] ["tidb lightning encountered error stack info"] [error="build local backend failed: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 172.24.xx.44:20360: i/o timeout\""] [errorVerbose="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 172.24.xx.44:20360: i/o timeout\"\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/juju_adaptor.go:15\ngithub.com/pingcap/br/pkg/lightning/backend/local.(*local).checkMultiIngestSupport\n\tgithub.com/pingcap/br/pkg/lightning/backend/local/local.go:944\ngithub.com/pingcap/br/pkg/lightning/backend/local.NewLocalBackend\n\tgithub.com/pingcap/br/pkg/lightning/backend/local/local.go:918\ngithub.com/pingcap/br/pkg/lightning/restore.NewRestoreControllerWithPauser\n\tgithub.com/pingcap/br/pkg/lightning/restore/restore.go:279\ngithub.com/pingcap/br/pkg/lightning/restore.NewRestoreController\n\tgithub.com/pingcap/br/pkg/lightning/restore/restore.go:220\ngithub.com/pingcap/br/pkg/lightning.(*Lightning).run\n\tgithub.com/pingcap/br/pkg/lightning/lightning.go:305\ngithub.com/pingcap/br/pkg/lightning.(*Lightning).RunOnce\n\tgithub.com/pingcap/br/pkg/lightning/lightning.go:192\nmain.main.func2\n\tcommand-line-arguments/main.go:86\nmain.main\n\tcommand-line-arguments/main.go:87\nruntime.main\n\truntime/proc.go:225\nruntime.goexit\n\truntime/asm_amd64.s:1371\nbuild local backend failed"]
[2021/07/07 13:52:18.583 +08:00] [ERROR] [main.go:91] ["tidb lightning encountered error"] [error="build local backend failed: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 172.24.xx.44:20360: i/o timeout\""]
```
tikv log no error，network no error，ping no error，0.0xxxms delay

i3 16xlarge, Every four disks are used as sort and dump paths

The IP that reported the error on the lightning server tenel tikv, ok

Any tikv-ctl execution is successful.

This problem recurs stably, the tikv of rpc failure is relatively random, distributed in different az

tidb version：v5.0.3
tidb-lightning: v5.1.0
dumpling: v4.0.13（dump success）

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
- If check multi ingest failed, fallback to set `supportMultiIngest=false` instead of directly return an error. close #1323

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release note

 - No release note

### CODE REVIEW

File: pkg/lightning/backend/local/local.go [hunk 1]
```diff
@@ -990,20 +990,43 @@ func (local *local) checkMultiIngestSupport(ctx context.Context, pdClient pd.Cli
...
 					local.supportMultiIngest = false
 					return nil
 				}
 			}
-			return errors.Trace(err)
+			log.L().Warn("check multi ingest support failed", zap.Error(err), zap.String("store", s.Address),
+				zap.Int("retry", i))
+		}
+		if err != nil {
+			log.L().Warn("check multi failed all retry, fallback to false")
```
Comment (kennytm) [inline pkg/lightning/backend/local/local.go:1027]: log the error too?
```suggestion
			log.L().Warn("check multi failed all retry, fallback to false", log.ShortError(err))
```

**Review by kennytm — APPROVED** [review]

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Pull request merged**